Spigot-API-Patches/0294-Add-PlayerGlassBottleFillEvent.patch

```diff
@@ -0,0 +1,109 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Dmitry Sidorov <jonmagon@gmail.com>
+Date: Mon, 3 May 2021 10:56:53 +0300
+Subject: [PATCH] Add PlayerGlassBottleFillEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerGlassBottleFillEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerGlassBottleFillEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..0fbd2262e0b0d506f00758590ff49588eae6ebe4
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerGlassBottleFillEvent.java
+@@ -0,0 +1,97 @@
++package com.destroystokyo.paper.event.player;
++
++import org.bukkit.Material;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.player.PlayerEvent;
++import org.bukkit.inventory.ItemStack;
++import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable;
++
++/**
++ * Called when a player fills a bottle with water, honey or dragon's breath.
++ */
++public class PlayerGlassBottleFillEvent extends PlayerEvent implements Cancellable {
++    private static final HandlerList handlers = new HandlerList();
++
++    @NotNull
++    private final ItemStack item;
++    @NotNull
++    private ItemStack newitem;
++
++    private boolean cancelled;
++
++    public PlayerGlassBottleFillEvent(@NotNull final Player player, @NotNull final ItemStack item,
++                                      @NotNull final ItemStack newitem) {
++        super(player);
++        this.item = item;
++        this.newitem = newitem;
++    }
++
++    @Override
++    public boolean isCancelled() {
++        return cancelled;
++    }
++
++    /**
++     * Set whether or not to cancel filling. If canceled,
++     * the bottle will not be filled.
++     *
++     * If set to false, it's recommended that a call to {@link Player#updateInventory()}
```

Comment:
`true` not `false`, but the inventory update should be done in the server in case of cancellation, not put away for plugins to bump into every time

Reply:
@JRoy in https://github.com/PaperMC/Paper/pull/5152#discussion_r568834399 says it shouldn't.

Reply:
having to cal updateInventory manually is a failure of API impl

Reply:
A little time has passed and already the opinions are different, okay I'll restore the very first version of the PR